lib/orderbook/OrderBook.ts

```diff
@@ -532,37 +532,40 @@ class OrderBook extends EventEmitter {
     const { r_hash, proposedQuantity, orderId, pairId } = requestPacket.body!;
 
     const order = this.tryGetOwnOrder(orderId, pairId);
-    if (order) {
-      const availableQuantity = order.hold ? order.quantity - order.hold : order.quantity;
-      // TODO: accept the smaller of the proposed quantity and the available quantity
-      if (availableQuantity >= proposedQuantity) {
-        // put accepted quantity on hold
-        // quantityToAccept = Math.min(proposedQuantity, availableQuantity);
-        const quantityToAccept = proposedQuantity;
-
-        order.hold = order.hold ? order.hold + quantityToAccept : quantityToAccept;
-
-        // try to accept the deal
-        const orderToAccept = {
-          quantityToAccept,
-          localId: order.localId,
-          price: order.price,
-        };
-        const dealAccepted = await this.swaps!.acceptDeal(orderToAccept, requestPacket, peer);
-        if (!dealAccepted) {
-          // release hold amount and reject swap
-          order.hold -= quantityToAccept;
-        }
-      } else {
-        peer.sendPacket(new SwapErrorPacket({
-          r_hash,
-          errorMessage: SwapFailureReason[SwapFailureReason.OrderUnavailable],
-        }, requestPacket.header.id));
+    if (!order) {
+      peer.sendPacket(new SwapErrorPacket({
+        r_hash,
+        errorMessage: SwapFailureReason[SwapFailureReason.OrderNotFound],
+      }, requestPacket.header.id));
+      return;
+    }
+
+    const availableQuantity = order.hold ? order.quantity - order.hold : order.quantity;
+    // TODO: accept the smaller of the proposed quantity and the available quantity
+    if (availableQuantity >= proposedQuantity) {
+      // put accepted quantity on hold
+      // quantityToAccept = Math.min(proposedQuantity, availableQuantity);
+      /** The quantity of the order that we will accept */
+      const quantity = proposedQuantity;
+
+      order.hold = order.hold ? order.hold + quantity : quantity;
```

Comment:
Consider making order.hold not nullable (initialize with zero).

It will make the code clearer 
>order.hold = order.hold + quantity;

Reply:
OK will do.